rplugin/python3/gdb/win.py

```diff
@@ -9,43 +10,24 @@ def __init__(self, vim, win, cursor, client, breakpoint, keymaps):
         self.keymaps = keymaps
 
     def jump(self, file, line):
-        # Make sure all the operations happen in the correct window
-        # TODO: Use api instead of Vim commands.
-        window = self.vim.current.window
-        mode = self.vim.api.get_mode()
-        if self.jumpWin != window:
-            # We're going to jump to another window and return.
-            # There is no need to change keymaps forth and back.
-            self.keymaps.setDispatchActive(False)
-            self.vim.command("%dwincmd w" % self.jumpWin.number)
-
         # Check whether the file is already loaded or load it
         targetBuf = self.vim.call("bufnr", file, 1)
 
-        # The terminal buffer may contain the name of the source file (in pdb, for
-        # instance)
-        if targetBuf == self.client.getBuf().handle:
-            self.vim.command("noswapfile view " + file)
-            targetBuf = self.vim.call("bufnr", file)
-
-        # Switch to the new buffer if necessary
-        if self.vim.call("bufnr", '%') != targetBuf:
-            self.vim.command('noswapfile buffer %d' % targetBuf)
+        # This file being opened having a .swp file causes this function to throw
+        try:
+            self.vim.call("nvim_win_set_buf", self.jumpWin.handle, targetBuf)
+        except NvimError as e:
+            pass
+        # TODO: figure out if other autocommands need ran here.
```

Comment:
The tests pass now, could you please rephrase the remark and get rid of TODO?

Reply:
The TODO is still valid. There's a few things that seem off. 

Somehow it's confusing other plugins that, I'm guessing, rely on the `autocommand`s vim regularly does to maintain synchronization. e.g. I'm getting coc.nvim throwing an error in the coc-python extension while debugging C++ code with lldb. So evidently some plugins are too confused by this. The neovim API functions life outside of the autocommand life cycle. So this might cause some headaches.

Second, when you first load a file it's unhighlighted for a second.

I'd like to play around with solutions for these two issues first.

Reply:
Does this mean that the change isn't ready for merging?

Reply:
Shrug, try it out if you'd like. The mode switch -> window switch -> buffer change -> window switch -> mode switch flow caused many issues for me (e.g. `(lldb) bt` had a habit of clobbering every file in the back trace with spammed text due to the mode switch done at the end of jump). I consider it a net upgrade even if buggy, but it's not quite done.

Reply:
Well. I've just tried the change, and I like it. It does work well when using keymaps `<f10>`, `<f11>`. But when I type GDB, LLDB commands in the terminal window, the screen isn't updating until I exit the terminal mode with `<esc>`. I believe this must be fixed before merging.

Reply:
The screen isn't updating? What do you mean? My screen updates exactly as I expect it to.

Reply:
Behold:
[![asciicast](https://asciinema.org/a/248398.svg)](https://asciinema.org/a/248398)